lua/hbac/init.lua

```diff
@@ -82,17 +82,58 @@ M.setup = function(opts)
 end
 
 
-local function close_unused()
+M.close_unpinned = function()
 	local curbufnr = vim.api.nvim_get_current_buf()
 	local buflist = vim.api.nvim_list_bufs()
 	for _, bufnr in ipairs(buflist) do
-		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.persistant_buffers[bufnr] then
+		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.pinned_buffers[bufnr] then
 			vim.cmd('bd ' .. tostring(bufnr))
 		end
+		print("Hbac: Closed unpinned buffers")
 	end
 end
 
+
+M.pin = function(bufnr)
+  M.pin_buffer(bufnr)
+  print("Hbac: Buffer pinned")
+end
+
+M.unpin = function(bufnr)
+	bufnr = bufnr or vim.api.nvim_get_current_buf()
+	M.pinned_buffers[bufnr] = nil
+  print("Hbac: Buffer unpinned")
+end
+
+local commands = {
+	pin = M.pin,
+	unpin = M.unpin,
+	close_unpinned = M.close_unpinned,
+}
+
+function M.hbac_cmd(arg)
+	if commands[arg] then
+		commands[arg]()
+	else
+		print("Unknown Hbac command:", arg)
+	end
+end
+
+vim.api.nvim_create_user_command("Hbac", function(args)
+	M.hbac_cmd(args.args)
+end, {
+	nargs = 1,
+	complete = function()
+		return { unpack(vim.tbl_keys(commands)) }
+	end,
+	}
+)
+
 return {
-	close_unused = close_unused,
+  pin = M.pin,
+  unpin = M.unpin,
+	close_unpinned = M.close_unpinned,
 	setup = M.setup,
+	hbac_cmd = M.hbac_cmd,
 }
```

Comment:
With a bigger refactor redefining `M` and redefining what functions are local vs. part of `M`, this could just return `M` and only make the necessary functions available to the user. If you think this is getting too messy.

Reply:
I agree, everything exported should be part of M, everything else can be local. That way the M.pin becomes the user command that triggers notification and local pin just pins the buffer.